servicetalk-opentracing-zipkin-publisher/src/main/java/io/servicetalk/opentracing/zipkin/publisher/reporter/HttpReporter.java

```diff
@@ -177,6 +179,13 @@ public Completable closeAsyncGracefully() {
         }
         return encodedSpans -> client.request(
                 client.post(path).addHeader(CONTENT_TYPE, contentType).payloadBody(encodedSpans))
+                .beforeOnSuccess(response -> {
+                    HttpResponseStatus status = response.status();
+                    if (status.statusClass() != SUCCESSFUL_2XX) {
+                        LOGGER.info("Unexpected response from the collector. Response headers: {}",
+                                response.toString((__, headerValue) -> headerValue));
```

Comment:
That may log any secret header values (auth tokens). Consider not logging all the headers.

Reply:
This is for response and there is a trade-off b/w this log being of limited usefulness and may  disclose more than necessary information. Sending secrets in a response header is somewhat less likely as compared to providing more information about the error in a header. So, I would like to keep this log here to aid debuggability.